packages/clay-css/src/scss/cadmin/components/_treeview.scss

```diff
@@ -0,0 +1,324 @@
+.treeview {
+	@include clay-css($cadmin-treeview);
+
+	.btn {
+		$btn: setter(map-get($cadmin-treeview, btn), ());
+
+		@include clay-button-variant($btn);
+	}
+
+	.btn-monospaced {
+		$btn-monospaced: setter(map-get($cadmin-treeview, btn-monospaced), ());
+
+		@include clay-button-variant($btn-monospaced);
+	}
+
+	.custom-control {
+		$custom-control: setter(map-get($cadmin-treeview, custom-control), ());
+
+		@include clay-css($custom-control);
+	}
+}
+
+.treeview-group {
+	$group: setter(map-get($cadmin-treeview, group), ());
+
+	@include clay-css($group);
+}
+
+.treeview-item {
+	$item: setter(map-get($cadmin-treeview, item), ());
+
+	@include clay-css($item);
+
+	&:last-child {
+		$last-child: setter(map-get($item, last-child), ());
+
+		@include clay-css($last-child);
+	}
+}
+
+.treeview-link {
+	$link: setter(map-get($cadmin-treeview, link), ());
+
+	@include clay-link($link);
+
+	&.hover,
+	&:hover,
+	&.focus,
+	&:focus {
+		.treeview-action-btn {
+			display: block;
+		}
+	}
+
+	&.show,
+	&[aria-expanded='true'] {
+		.treeview-expander-btn {
+			.treeview-expanded-d-none {
+				display: none;
+			}
+
+			.lexicon-icon:not(.treeview-expanded-d-none) {
+				display: inline-block;
+			}
+		}
+	}
+}
+
+.treeview-action-btn {
+	$action-btn: setter(map-get($cadmin-treeview, action-btn), ());
+
+	@include clay-button-variant($action-btn);
+}
+
+.treeview-expander-btn {
+	$expander-btn: setter(map-get($cadmin-treeview, expander-btn), ());
+
+	@include clay-button-variant($expander-btn);
+
+	.lexicon-icon:not(.treeview-expanded-d-none) {
+		display: none;
+	}
+}
+
+.treeview-icon {
+	$icon: setter(map-get($cadmin-treeview, icon), ());
+
+	@include clay-css($icon);
+}
+
+.treeview-text {
+	$text: setter(map-get($cadmin-treeview, text), ());
+
+	@include clay-css($text);
+}
+
+.treeview-nested {
```

Comment:
`treeview-nested` goes 4/5 levels depending where level 1 starts. I can't add N levels because the CSS gets too large. This is due to using padding to create the nesting. We can create N levels using margin without large CSS, but requires some pseudo element hacks. I wanted to leave those open for end user customizations. I can look into this more.

Reply:
Hmm, ok, I think understand, If we need N levels, we add `.treeview-nested` in the depth? Can we maybe move the padding to `.treeview-group`, as it's being nested, adding `24px`, or would that be limiting because of hovering on the items?